api/src/main/java/org/apache/iceberg/transforms/Timestamps.java

```diff
@@ -89,17 +153,21 @@ public SerializableFunction<Long, Integer> bind(Type type) {
 
   @Override
   public boolean canTransform(Type type) {
-    return type.typeId() == Type.TypeID.TIMESTAMP;
+    return type.typeId() == Type.TypeID.TIMESTAMP || type.typeId() == Type.TypeID.TIMESTAMP_NANO;
   }
 
   @Override
   public Type getResultType(Type sourceType) {
-    if (granularity == ChronoUnit.DAYS) {
+    if (apply.resultTypeUnit == ChronoUnit.DAYS) {
       return Types.DateType.get();
     }
     return Types.IntegerType.get();
   }
 
+  public ChronoUnit getResultTypeUnit() {
```

Comment:
Is it possible to avoid exposing `ChronoUnit`?

Also, Iceberg method names should almost never use `get`. Either use a more specific verb or omit `get` because it doesn't add anything.

Reply:
Done and done.